agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/AiComponentInstaller.java

```diff
@@ -313,11 +316,23 @@ else if (!jmxMetric.attributes.isEmpty()) {
     //assuming that every other element is not needed - such as "type", "worker", "name"
     //assuming same set of delimiters in every object name
     static String getJmxDisplayName(String attribute) {
-        String[] nameParts = attribute.split("=|:|,");
-        StringBuilder name = new StringBuilder();
 
-        for(int i=0; i<nameParts.length;i=i+2) {
-            name.append(nameParts[i]).append(" / ");
+        StringBuilder name = new StringBuilder("");
+        try
+        {
+            ObjectName nameObject = new ObjectName(attribute);
+            name.append(nameObject.getDomain()).append(" / ");
+            Hashtable properties = nameObject.getKeyPropertyList();
+
+            if(properties.containsKey("type")) {
+                name.append(properties.get("type")).append(" / ");
+            }
+            if(properties.containsKey("name")) {
+                name.append(properties.get("name")).append(" / ");
+            }
```

Comment:
only handling `type` and `name` will lead to duplicate names for mbean names that vary only on other fields (which is common)

what do you think of simplifying, and just appending the full objectName + attributeName? it's not as pretty, but would require less documentation/explanation for users

Reply:
@trask 
Just like in Glowroot? 

For example:
1.
"Catalina:type=RequestProcessor,worker="http-nio-8080",name=HttpRequest1"
would be displayed as:
Catalina / RequestProcessor / "http-nio-8080" / HttpRequest1 / (plus attribute name here)

2. "tomcat.jdbc:name=\"jdbc/storage\",context=/,engine=Catalina,type=ConnectionPool,host=localhost,class=org.apache.tomcat.jdbc.pool.DataSource"

Would be displayed as:

"tomcat.jdbc / "jdbc/storage" / / / Catalina / ConnectionPool / localhost / org.apache.tomcat.jdbc.pool.DataSource / (plus attribute name here)

Please confirm

Reply:
no, suggesting potentially simpler:

Catalina:type=RequestProcessor,worker="http-nio-8080",name=HttpRequest1 (plus attribute name here)

Reply:
@trask 
I changed the naming algorithm up again. See what you think.
Will generate names as:
Catalina /RequestProcessor /"http-nio-8080" /HttpRequest1 / (plus attribute name here)

I can always revert to the simple way as you suggested earlier. Or I can implement an algorithm like you did in GlowRoot, to identify the beginning and end of the quoted property (such as "http-nio-8080" in the object name above)